src/Microsoft.AspNet.Security.OAuth/OAuthBearerAuthenticationOptions.cs

```diff
@@ -33,34 +44,121 @@ public OAuthBearerAuthenticationOptions() : base()
-        /// provider or format is assigned, a compatible instance must be assigned to the OAuthAuthorizationServerOptions.AccessTokenProvider 
-        /// and OAuthAuthorizationServerOptions.AccessTokenFormat of the authorization server.
+        /// Gets or sets the timeout when using the backchannel to make an http call.
         /// </summary>
-        public IAuthenticationTokenProvider AccessTokenProvider { get; set; }
+        public TimeSpan BackchannelTimeout { get; set; }
 
+#if ASPNET50
         /// <summary>
-        /// Used to know what the current clock time is when calculating or validating token expiration. When not assigned default is based on
-        /// DateTimeOffset.UtcNow. This is typically needed only for unit testing.
+        /// Gets or sets the a pinned certificate validator to use to validate the endpoints used
+        /// when retrieving metadata.
         /// </summary>
-        public ISystemClock SystemClock { get; set; }
-    }
+        /// <value>
+        /// The pinned certificate validator.
+        /// </value>
+        /// <remarks>If this property is null then the default certificate checks are performed,
+        /// validating the subject name and if the signing chain is a trusted party.</remarks>
+        public ICertificateValidator BackchannelCertificateValidator { get; set; }
+#endif
+		/// <summary>
+		/// Configuration provided directly by the developer. If provided, then MetadataAddress and the Backchannel properties
+		/// will not be used. This information should not be updated during request processing.
+		/// </summary>
+		public OpenIdConnectConfiguration Configuration { get; set; }
+
+		/// <summary>
+		/// Responsible for retrieving, caching, and refreshing the configuration from metadata.
+		/// If not provided, then one will be created using the MetadataAddress and Backchannel properties.
+		/// </summary>
+		public IConfigurationManager<OpenIdConnectConfiguration> ConfigurationManager { get; set; }
+
+		/// <summary>
+		/// Gets or sets if a metadata refresh should be attempted after a SecurityTokenSignatureKeyNotFoundException. This allows for automatic
+		/// recovery in the event of a signature key rollover. This is enabled by default.
+		/// </summary>
+		public bool RefreshOnIssuerKeyNotFound { get; set; }
+
+		/// <summary>
+		/// Used to know what the current clock time is when calculating or validating token expiration. When not assigned default is based on
+		/// DateTimeOffset.UtcNow. This is typically needed only for unit testing.
+		/// </summary>
+		public ISystemClock SystemClock { get; set; }
+
+		/// <summary>
+		/// Gets or sets the <see cref="SecurityTokenValidators"/> of <see cref="SecurityTokenHandler"/>s used to read and validate <see cref="SecurityToken"/>s. 
+		/// </summary>
+		/// <exception cref="ArgumentNullException">if 'value' is null.</exception>
+		public ICollection<ISecurityTokenValidator> SecurityTokenValidators
```

Comment:
It's a best practice to make collection properties non-settable:

http://msdn.microsoft.com/en-us/library/ms182327(v=vs.120).aspx
http://msdn.microsoft.com/en-us/library/dn169389%28v=vs.110%29.aspx